ejerciciosPilas/src/escenas/libroPrimaria/EscenaCoty.ts

```diff
@@ -3,81 +3,82 @@
 /// <reference path = "../../actores/libroPrimaria/Charco.ts" />
 /// <reference path = "../../actores/FlechaEscenarioAleatorio.ts" />
 
-type ArgumentosCoty = {xCoty?: number, yCoty?: number, longitudSegmento?: number, puedeHaberCharco?: boolean}
-
- class EscenaCoty extends DibujandoFiguras {
-   _puntosEsperados: PuntoSimple[] | PuntoSimple[][];
-   dibujoPreexistente: DibujoLineal;
-   pizarraDibujoPreexistente: Pizarra;
-   xCoty: number;
-   yCoty: number;
-   charco: Charco;
-   puedeHaberCharco: boolean;
-   longitudSegmento: number;
-
-   constructor(dibujoPreexistente: PuntoSimple[] | PuntoSimple[][] = [], puntosEsperados: PuntoSimple[] | PuntoSimple[][] = [], argumentos: ArgumentosCoty){
-     super();
-     this._puntosEsperados = puntosEsperados;
-     this.dibujoPreexistente = DibujoLineal.desdePuntosSimples(dibujoPreexistente);
-     this.sanitizarArgumentos(argumentos);
-   }
-
-    sanitizarArgumentos(argumentos: ArgumentosCoty) {
-      this.xCoty = argumentos.xCoty || 0;
-      this.yCoty = argumentos.yCoty || 0;
-      this.longitudSegmento = argumentos.longitudSegmento || 50;
-      this.puedeHaberCharco = Boolean(argumentos.puedeHaberCharco);
+type ArgumentosCoty = { xCoty?: number, yCoty?: number, longitudSegmento?: number, puedeHaberCharco?: boolean }
+
+class EscenaCoty extends DibujandoFiguras {
+  _puntosEsperados: PuntoSimple[] | PuntoSimple[][];
+  dibujoPreexistente: DibujoLineal;
+  pizarraDibujoPreexistente: Pizarra;
+  xCoty: number;
+  yCoty: number;
+  charco: Charco;
+  puedeHaberCharco: boolean;
+  longitudSegmento: number;
+
+  constructor(dibujoPreexistente: PuntoSimple[] | PuntoSimple[][] = [], puntosEsperados: PuntoSimple[] | PuntoSimple[][] = [], argumentos: ArgumentosCoty) {
+    super();
+    this._puntosEsperados = puntosEsperados;
+    this.dibujoPreexistente = DibujoLineal.desdePuntosSimples(dibujoPreexistente);
+    this.sanitizarArgumentos(argumentos);
+  }
+
+  sanitizarArgumentos(argumentos: ArgumentosCoty) {
+    this.xCoty = argumentos.xCoty || 0;
+    this.yCoty = argumentos.yCoty || 0;
+    this.longitudSegmento = argumentos.longitudSegmento || 50;
+    this.puedeHaberCharco = Boolean(argumentos.puedeHaberCharco);
+  }
+
+  iniciar() {
+    if (this.puedeHaberCharco) {
+      new FlechaEscenarioAleatorio();
+    }
+    if (this.puedeHaberCharco && Math.random() >= 0.5) {
+      this.crearCharco()
```

Comment:
Ya se que esto ya estaba, pero este `if` repite lógica con el de arriba, podría estar dentro.

Reply:
Yyy, no se si estoy en lo correcto, pero creo que se podría mover todo después de la llama a `super` y así unificar con el `this.ubicarCharco();` y sacarnos un `if` de encima.